pkg/artifactory/resource_artifactory_pull_replication.go

```diff
@@ -0,0 +1,115 @@
+package artifactory
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+)
+
+func resourceArtifactoryPullReplication() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourcePullReplicationCreate,
+		ReadContext:   resourcePullReplicationRead,
+		UpdateContext: resourcePullReplicationUpdate,
+		DeleteContext: resourceReplicationDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema:      mergeSchema(replicationSchemaCommon, replicationSchema),
+		Description: "Used for configuring pull replication on remote repos.",
+	}
+}
+
+func unpackPullReplication(s *schema.ResourceData) *utils.ReplicationBody {
+	d := &ResourceData{s}
+	replicationConfig := new(utils.ReplicationBody)
+
+	replicationConfig.RepoKey = d.getString("repo_key", false)
+	replicationConfig.CronExp = d.getString("cron_exp", false)
+	replicationConfig.EnableEventReplication = d.getBool("enable_event_replication", false)
+	replicationConfig.Enabled = d.getBool("enabled", false)
+	replicationConfig.SyncDeletes = d.getBool("sync_deletes", false)
+	replicationConfig.SyncProperties = d.getBool("sync_properties", false)
+	replicationConfig.SyncStatistics = d.getBool("sync_statistics", false)
+	replicationConfig.PathPrefix = d.getString("path_prefix", false)
+
+	return replicationConfig
+}
+
+func packPullReplicationBody(config PullReplication, d *schema.ResourceData) diag.Diagnostics {
```

Comment:
So, the `universalPack` function actually does this magic for you and you should use it. It will snake case the properties of the struct and use those for the `hcl` name - you can override with an additional struct tag of `hcl:"foo_tag"`. Not requiring this, but we are trying to move away from hard coding